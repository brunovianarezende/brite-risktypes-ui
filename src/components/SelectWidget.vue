<template>
  <el-form-item :label="label">
    <el-select :value="value" @input="value => { $emit('input', value) }">
      <el-option
        v-for="item in options"
        :key="item"
        :label="item"
        :value="item">
      </el-option>
    </el-select>
  </el-form-item>
</template>

<script>
export default {
  name: 'SelectWidget',
  props: ['value', 'label', 'options']
}
</script>
