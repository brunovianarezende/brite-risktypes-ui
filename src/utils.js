import Vue from 'vue'
import ElementUI from 'element-ui'
import locale from 'element-ui/lib/locale/lang/en'
import VModal from 'vue-js-modal'

export const registerGlobalComponents = () => {
  Vue.use(ElementUI, { locale })
  Vue.use(VModal)
}
