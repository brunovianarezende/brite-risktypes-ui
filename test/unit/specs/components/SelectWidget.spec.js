import Vue from 'vue'
import SelectWidget from '@/components/SelectWidget'
import {registerGlobalComponents} from '@/utils'
import { triggerEvent } from './utils'

registerGlobalComponents()

describe('Select.vue', () => {
  describe('interface', () => {
    it('should render correctly', () => {
      const Comp = {
        render (createElement) {
          return createElement(
            'el-form', {}, [
              createElement(SelectWidget,
                {props: {label: this.label, options: this.options, value: ''}})
            ]
          )
        },
        props: ['label', 'options'],
        components: {SelectWidget},
        propsData: {
          label: 'The label',
          options: ['a', 'B', 'c']
        }
      }

      const vm = new Vue(Comp).$mount()
      return Vue.nextTick()
        .then(() => {
          expect(vm.$el).toMatchSnapshot()
        })
    })

    it('should support v-model', () => {
      const Comp = {
        render (createElement) {
          const div = createElement('div', this.value)
          const selectWidget = createElement(SelectWidget,
            {
              props: {label: this.label, options: this.options, value: this.value},
              on: { input: (v) => { this.value = v } }
            })

          return createElement(
            'el-form', {}, [div, selectWidget]
          )
        },
        props: ['label', 'options'],
        data: {
          value: 'c'
        },
        components: {SelectWidget},
        propsData: {
          label: 'The label',
          options: ['a', 'B', 'c']
        }
      }

      const vm = new Vue(Comp).$mount()
      return Vue.nextTick()
        .then(() => {
          expect(vm.$el).toMatchSnapshot()
          const options = vm.$el.querySelectorAll('.el-select-dropdown__item')
          triggerEvent(options[1], 'mouseenter')
          options[1].click()
          return Vue.nextTick()
        })
        .then(() => {
          expect(vm.$el).toMatchSnapshot()
        })
    })
  })
})
