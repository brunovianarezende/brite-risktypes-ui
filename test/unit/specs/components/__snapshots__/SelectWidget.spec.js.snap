// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select.vue interface should render correctly 1`] = `
<form
  class="el-form"
>
  <div
    class="el-form-item"
  >
    <label
      class="el-form-item__label"
    >
      The label
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-select"
      >
        <!---->
        <div
          class="el-input el-input--suffix"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            placeholder="Select"
            readonly="readonly"
            rows="2"
            size=""
            type="text"
          />
          <!---->
          <span
            class="el-input__suffix"
          >
            <span
              class="el-input__suffix-inner"
            >
              <i
                class="el-select__caret el-input__icon el-icon-arrow-up"
              />
              <!---->
            </span>
            <!---->
          </span>
          <!---->
        </div>
        <div
          class="el-select-dropdown el-popper"
          style="display: none;"
        >
          <div
            class="el-scrollbar"
            style=""
          >
            <div
              class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default"
            >
              <ul
                class="el-scrollbar__view el-select-dropdown__list"
              >
                <!---->
                <li
                  class="el-select-dropdown__item"
                >
                  <span>
                    a
                  </span>
                </li>
                <li
                  class="el-select-dropdown__item"
                >
                  <span>
                    B
                  </span>
                </li>
                <li
                  class="el-select-dropdown__item"
                >
                  <span>
                    c
                  </span>
                </li>
                <div
                  class="resize-triggers"
                >
                  <div
                    class="expand-trigger"
                  >
                    <div
                      style="width: 1px; height: 1px;"
                    />
                  </div>
                  <div
                    class="contract-trigger"
                  />
                </div>
              </ul>
            </div>
            <div
              class="el-scrollbar__bar is-horizontal"
            >
              <div
                class="el-scrollbar__thumb"
                style="-webkit-transform: translateX(0%);"
              />
            </div>
            <div
              class="el-scrollbar__bar is-vertical"
            >
              <div
                class="el-scrollbar__thumb"
                style="-webkit-transform: translateY(0%);"
              />
            </div>
          </div>
          <!---->
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Select.vue interface should support v-model 1`] = `
<form
  class="el-form"
>
  <div>
    c
  </div>
  <div
    class="el-form-item"
  >
    <label
      class="el-form-item__label"
    >
      The label
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-select"
      >
        <!---->
        <div
          class="el-input el-input--suffix"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            placeholder="Select"
            readonly="readonly"
            rows="2"
            size=""
            type="text"
          />
          <!---->
          <span
            class="el-input__suffix"
          >
            <span
              class="el-input__suffix-inner"
            >
              <i
                class="el-select__caret el-input__icon el-icon-arrow-up"
              />
              <!---->
            </span>
            <!---->
          </span>
          <!---->
        </div>
        <div
          class="el-select-dropdown el-popper"
          style="display: none;"
        >
          <div
            class="el-scrollbar"
            style=""
          >
            <div
              class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default"
            >
              <ul
                class="el-scrollbar__view el-select-dropdown__list"
              >
                <!---->
                <li
                  class="el-select-dropdown__item"
                >
                  <span>
                    a
                  </span>
                </li>
                <li
                  class="el-select-dropdown__item"
                >
                  <span>
                    B
                  </span>
                </li>
                <li
                  class="el-select-dropdown__item selected"
                >
                  <span>
                    c
                  </span>
                </li>
                <div
                  class="resize-triggers"
                >
                  <div
                    class="expand-trigger"
                  >
                    <div
                      style="width: 1px; height: 1px;"
                    />
                  </div>
                  <div
                    class="contract-trigger"
                  />
                </div>
              </ul>
            </div>
            <div
              class="el-scrollbar__bar is-horizontal"
            >
              <div
                class="el-scrollbar__thumb"
                style="-webkit-transform: translateX(0%);"
              />
            </div>
            <div
              class="el-scrollbar__bar is-vertical"
            >
              <div
                class="el-scrollbar__thumb"
                style="-webkit-transform: translateY(0%);"
              />
            </div>
          </div>
          <!---->
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`Select.vue interface should support v-model 2`] = `
<form
  class="el-form"
>
  <div>
    B
  </div>
  <div
    class="el-form-item"
  >
    <label
      class="el-form-item__label"
    >
      The label
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-select"
      >
        <!---->
        <div
          class="el-input el-input--suffix"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            placeholder="Select"
            readonly="readonly"
            rows="2"
            size=""
            type="text"
          />
          <!---->
          <span
            class="el-input__suffix"
          >
            <span
              class="el-input__suffix-inner"
            >
              <i
                class="el-select__caret el-input__icon el-icon-arrow-up"
              />
              <!---->
            </span>
            <!---->
          </span>
          <!---->
        </div>
        <div
          class="el-select-dropdown el-popper"
          style="display: none;"
        >
          <div
            class="el-scrollbar"
            style=""
          >
            <div
              class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default"
            >
              <ul
                class="el-scrollbar__view el-select-dropdown__list"
              >
                <!---->
                <li
                  class="el-select-dropdown__item"
                >
                  <span>
                    a
                  </span>
                </li>
                <li
                  class="el-select-dropdown__item selected hover"
                >
                  <span>
                    B
                  </span>
                </li>
                <li
                  class="el-select-dropdown__item"
                >
                  <span>
                    c
                  </span>
                </li>
                <div
                  class="resize-triggers"
                >
                  <div
                    class="expand-trigger"
                  >
                    <div
                      style="width: 1px; height: 1px;"
                    />
                  </div>
                  <div
                    class="contract-trigger"
                  />
                </div>
              </ul>
            </div>
            <div
              class="el-scrollbar__bar is-horizontal"
            >
              <div
                class="el-scrollbar__thumb"
                style="-webkit-transform: translateX(NaN%);"
              />
            </div>
            <div
              class="el-scrollbar__bar is-vertical"
            >
              <div
                class="el-scrollbar__thumb"
                style="-webkit-transform: translateY(NaN%);"
              />
            </div>
          </div>
          <!---->
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
      <!---->
    </div>
  </div>
</form>
`;
